odpf/optimus/core/v1beta1/runtime.proto

```diff
@@ -559,13 +575,40 @@ message RegisterSecretRequest {
     string project_name = 1;
     string secret_name = 2;
     string value = 3; // base64 encoded secret value
+    string namespace_name = 4;
 }
 
 message RegisterSecretResponse {
     bool success = 1;
     string message = 2;
 }
 
+message UpdateSecretRequest {
+    string project_name = 1;
+    string secret_name = 2;
+    string value = 3; // base64 encoded secret value
+    string namespace_name = 4;
+}
+
+message UpdateSecretResponse {
+    bool success = 1;
+    string message = 2;
+}
+
+message ListSecretsRequest {
+    string project_name = 1;
+}
+
+message ListSecretsResponse {
+    message Secret {
```

Comment:
I don't see we have a `Get` call for secrets yet. I guess we will have it soon? And when we do are we going to create a separate message for it?

Reply:
The secret message will be different in that case, that will require name and value of secret.